### PR TITLE
New serverless pattern  - lambda cloudwatch metrics and dashboard in cdk - dotnet

### PR DESCRIPTION
**Pull request by srcsakthivel**
*Description of changes:*

This pattern helps build CloudWatch Dashboard with AWS Lambda Metrics. The Dashboard built with 4 widgets,
- Invocations - from AWS/Lambda Namespace
- memory_utilization - from LambdaInsights Namespace
- cpu_total_time - from LambdaInsights Namespace
- Custom Metrics (Proxy-Request & Proxy-Successful) - from Custom Namespace

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cw-dashboard-lambda-metrics/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    dotnet restore src
+    ```
+4. Deploy the stack to your default AWS account and region.
+    ```
+    cdk deploy
+    ```
+
+## How it works
+
+Cloudwatch Dashboard created on top of the metrics produced by Lambda Function. Along with the default metrics in the default Namespace, Docker Lambda Function instrumented with Lambda Insight to get the Insight Metrics. Lambda uses Powertools to send the custom metrics. In Lambda function we have two custom metrics one for all the requests and one for successful.
```
Comment (mavi888) [inline cw-dashboard-lambda-metrics/README.md:44]: can you rewrite this paragraph to be more clear, there are a lot of things happening here and it is hard to read. You might need many paragraphs to explain this.

File: cw-dashboard-lambda-metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Cleanup
+ 
+Run the given command to delete the resources that were created. It might take some time for the CloudFormation stack to get deleted.
+```
+cdk destroy
+```
+
+----
+Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (mavi888) [inline cw-dashboard-lambda-metrics/README.md:82]: 2023

File: cw-dashboard-lambda-metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ```
+- Payload with error status_code with no response 
+
+    ```json
+    {
+        "requestUrl": "https://httpbin.org/status/500"
+    }
+    ```
+
+After running some transactions. Dashboard should start showing the data in graph.
```
Comment (mavi888) [inline cw-dashboard-lambda-metrics/README.md:72]: attach a screenshot of how the dashboard looks after testing

File: cw-dashboard-lambda-metrics/src/lambda/proxy-lambda/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+FROM build AS publish
+RUN dotnet publish "proxy-lambda.csproj" \
+            --configuration Release \ 
+            --runtime linux-x64 \
+            --self-contained false \ 
+            --output /app/publish \
+            -p:PublishReadyToRun=true  
+
+FROM base AS final
+RUN curl -O https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm && \
```
Comment (mavi888) [inline cw-dashboard-lambda-metrics/src/lambda/proxy-lambda/Dockerfile:21]: you are fixing the extension region here and if you deploy from a different region fails.you need to make this dynamic.

**Comment by mavi888** [comment]
@srcsakthivel there are some issues with this pattern. please fix them and then i can proceed with the review. 
Thanks a lot :)

File: cw-dashboard-lambda-metrics/README.md (hunk 1 above)
Comment (srcsakthivel) [reply cw-dashboard-lambda-metrics/README.md]: I have updated the para and added definition on individual items.

File: cw-dashboard-lambda-metrics/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    cdk deploy
+    ```
+
+## How it works
+
+Cloudwatch Dashboard created on top of the metrics produced by Lambda Function. Along with the default metrics in the default Namespace, Docker Lambda Function instrumented with Lambda Insight to get the Insight Metrics. Lambda uses Powertools to send the custom metrics. In Lambda function we have two custom metrics one for all the requests and one for successful. 
+
+## Testing
+
+Run some transactions from AWS Console - Lambda - Testing using the following events.
```
[the remark this answers is not in the document]
Comment (srcsakthivel) [reply cw-dashboard-lambda-metrics/README.md:62]: I have added the aws cli invoke command

File: cw-dashboard-lambda-metrics/README.md (hunk 2 above)
Comment (srcsakthivel) [reply cw-dashboard-lambda-metrics/README.md]: updated

File: cw-dashboard-lambda-metrics/src/lambda/proxy-lambda/Dockerfile (hunk 4 above)
Comment (srcsakthivel) [reply cw-dashboard-lambda-metrics/src/lambda/proxy-lambda/Dockerfile]: I have made it as a ARG in dockerfile and passing it from the Stack.

File: cw-dashboard-lambda-metrics/README.md (hunk 3 above)
Comment (srcsakthivel) [reply cw-dashboard-lambda-metrics/README.md:96]: added the screenshot.

**Comment by srcsakthivel** [comment]
> @srcsakthivel there are some issues with this pattern. please fix them and then i can proceed with the review. Thanks a lot :)

Thanks for the review and valuable feedback. I have made the updates and ready for your review.

**Comment by srcsakthivel** [comment]
Took sometime to reproduce the issue. I have added in the requirement section and troubleshooting step.

**Comment by mavi888** [comment]
This is still failing when following the instructions.

`````
cw-dashboard-lambda-metrics git:(srcsakthivel/main) ✗ cdk deploy

✨  Synthesis time: 9.32s

DotnetLambdaCloudwatchDashboardStack:  start: Building 98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53:current_account-current_region
#1 [internal] load build definition from Dockerfile
#1 sha256:83b199679e5b6e8d12e11023e0e45c37d6fc996c5fc7dd728ea4268f67b5163f
#1 transferring dockerfile: 950B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:648c7f39bf0d2dcd0eb69827f21ed1412c3c0695aa8dbd9c626652f489da9f3c
#2 transferring context: 2B done
#2 DONE 0.0s

#4 [internal] load metadata for public.ecr.aws/lambda/dotnet:6
#4 sha256:bc7065f21dee32ea7736ecb265d6ad759ce7f172f06256d480fe7154ca7f9074
#4 ...

#3 [internal] load metadata for mcr.microsoft.com/dotnet/sdk:6.0-bullseye-slim
#3 sha256:3958982a3cf434b4942b75846a4c6ed5d14bc6423ddd7d25d603df9ba4fd2d51
#3 DONE 0.3s

#4 [internal] load metadata for public.ecr.aws/lambda/dotnet:6
#4 sha256:bc7065f21dee32ea7736ecb265d6ad759ce7f172f06256d480fe7154ca7f9074
#4 DONE 0.7s

#5 [base 1/1] FROM public.ecr.aws/lambda/dotnet:6@sha256:820abb60697e4e6d3a8c62610a167148616c601cc9be6ba64626e857319e7eb9
#5 sha256:79996b429f899dd87208fde324b714639953df94bcfc51062dc41efc40927d59
#5 CACHED

#11 [internal] load build context
#11 sha256:7bb1ca1b4c60c007ce8c41e0ca78550b542e0ef67e49a284cd2d53243996ae37
#11 transferring context: 6.50kB 0.0s done
#11 DONE 0.0s

#9 [build 1/7] FROM mcr.microsoft.com/dotnet/sdk:6.0-bullseye-slim@sha256:fda5db18bc50cd935a424b08c0237f6bc9f9f9559f72e3076858454e0b59a4d5
#9 sha256:6eb9424c33071cb48aa448595f8b2e8c48ac986a9135ec43629c8db9f505b745
#9 resolve mcr.microsoft.com/dotnet/sdk:6.0-bullseye-slim@sha256:fda5db18bc50cd935a424b08c0237f6bc9f9f9559f72e3076858454e0b59a4d5 0.0s done
#9 sha256:fda5db18bc50cd935a424b08c0237f6bc9f9f9559f72e3076858454e0b59a4d5 1.11kB / 1.11kB done
#9 sha256:5468fcfb7b1376bed09389195d4753bba08539e5cf0a0e33c39af7e318486f51 2.01kB / 2.01kB done
#9 sha256:a9ff2c65eb3a5852745e26674f4c9ba218b2ac584bf7dd99ee6001de4d290693 7.23kB / 7.23kB done
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 0B / 30.05MB 0.1s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 0B / 15.12MB 0.1s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 0B / 29.31MB 0.1s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 2.10MB / 30.05MB 1.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 1.05MB / 15.12MB 1.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 2.10MB / 15.12MB 1.5s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 3.15MB / 15.12MB 1.7s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 4.19MB / 30.05MB 2.2s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 2.10MB / 29.31MB 2.2s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 4.19MB / 15.12MB 2.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 5.24MB / 15.12MB 2.8s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 6.29MB / 30.05MB 3.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 6.29MB / 15.12MB 3.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 7.34MB / 15.12MB 3.7s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 8.39MB / 15.12MB 3.8s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 4.19MB / 29.31MB 3.8s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 8.39MB / 30.05MB 4.0s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 9.44MB / 15.12MB 4.2s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 10.49MB / 15.12MB 4.5s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 10.49MB / 30.05MB 4.7s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 11.53MB / 15.12MB 4.9s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 6.29MB / 29.31MB 4.9s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 12.58MB / 15.12MB 5.3s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 12.58MB / 30.05MB 5.5s
#9 CANCELED

#6 [final 1/4] RUN curl -O https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm
#6 sha256:3d4cb91a064609bd3c81f84ea5ee081265336c45c2fc795064ea676ed182b978
#6 0.230   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#6 0.230                                  Dload  Upload   Total   Spent    Left  Speed
100 4071k  100 4071k    0     0   786k      0  0:00:05  0:00:05 --:--:--  858k
#6 5.479        package AWSLogsLambdaInsights-1.0-1.x86_64 is intended for a different architecture
#6 ERROR: executor failed running [/bin/sh -c curl -O $INSIGHTSEXTENSION &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm]: exit code: 1
------
 > [final 1/4] RUN curl -O https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm:
------
executor failed running [/bin/sh -c curl -O $INSIGHTSEXTENSION &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm]: exit code: 1
DotnetLambdaCloudwatchDashboardStack:  fail: docker build --build-arg INSIGHTSEXTENSION=https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm --tag cdkasset-98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53 . exited with error code 1: #1 [internal] load build definition from Dockerfile
#1 sha256:83b199679e5b6e8d12e11023e0e45c37d6fc996c5fc7dd728ea4268f67b5163f
#1 transferring dockerfile: 950B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:648c7f39bf0d2dcd0eb69827f21ed1412c3c0695aa8dbd9c626652f489da9f3c
#2 transferring context: 2B done
#2 DONE 0.0s

#4 [internal] load metadata for public.ecr.aws/lambda/dotnet:6
#4 sha256:bc7065f21dee32ea7736ecb265d6ad759ce7f172f06256d480fe7154ca7f9074
#4 ...

#3 [internal] load metadata for mcr.microsoft.com/dotnet/sdk:6.0-bullseye-slim
#3 sha256:3958982a3cf434b4942b75846a4c6ed5d14bc6423ddd7d25d603df9ba4fd2d51
#3 DONE 0.3s

#4 [internal] load metadata for public.ecr.aws/lambda/dotnet:6
#4 sha256:bc7065f21dee32ea7736ecb265d6ad759ce7f172f06256d480fe7154ca7f9074
#4 DONE 0.7s

#5 [base 1/1] FROM public.ecr.aws/lambda/dotnet:6@sha256:820abb60697e4e6d3a8c62610a167148616c601cc9be6ba64626e857319e7eb9
#5 sha256:79996b429f899dd87208fde324b714639953df94bcfc51062dc41efc40927d59
#5 CACHED

#11 [internal] load build context
#11 sha256:7bb1ca1b4c60c007ce8c41e0ca78550b542e0ef67e49a284cd2d53243996ae37
#11 transferring context: 6.50kB 0.0s done
#11 DONE 0.0s

#9 [build 1/7] FROM mcr.microsoft.com/dotnet/sdk:6.0-bullseye-slim@sha256:fda5db18bc50cd935a424b08c0237f6bc9f9f9559f72e3076858454e0b59a4d5
#9 sha256:6eb9424c33071cb48aa448595f8b2e8c48ac986a9135ec43629c8db9f505b745
#9 resolve mcr.microsoft.com/dotnet/sdk:6.0-bullseye-slim@sha256:fda5db18bc50cd935a424b08c0237f6bc9f9f9559f72e3076858454e0b59a4d5 0.0s done
#9 sha256:fda5db18bc50cd935a424b08c0237f6bc9f9f9559f72e3076858454e0b59a4d5 1.11kB / 1.11kB done
#9 sha256:5468fcfb7b1376bed09389195d4753bba08539e5cf0a0e33c39af7e318486f51 2.01kB / 2.01kB done
#9 sha256:a9ff2c65eb3a5852745e26674f4c9ba218b2ac584bf7dd99ee6001de4d290693 7.23kB / 7.23kB done
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 0B / 30.05MB 0.1s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 0B / 15.12MB 0.1s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 0B / 29.31MB 0.1s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 2.10MB / 30.05MB 1.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 1.05MB / 15.12MB 1.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 2.10MB / 15.12MB 1.5s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 3.15MB / 15.12MB 1.7s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 4.19MB / 30.05MB 2.2s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 2.10MB / 29.31MB 2.2s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 4.19MB / 15.12MB 2.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 5.24MB / 15.12MB 2.8s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 6.29MB / 30.05MB 3.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 6.29MB / 15.12MB 3.3s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 7.34MB / 15.12MB 3.7s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 8.39MB / 15.12MB 3.8s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 4.19MB / 29.31MB 3.8s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 8.39MB / 30.05MB 4.0s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 9.44MB / 15.12MB 4.2s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 10.49MB / 15.12MB 4.5s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 10.49MB / 30.05MB 4.7s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 11.53MB / 15.12MB 4.9s
#9 sha256:65fd06d6030b0a851846fb12313f34355edb6df82e1e20e6cccc183c6e2196e8 6.29MB / 29.31MB 4.9s
#9 sha256:b8bd4d0c9c3231cb70f0074a09e8af9dd4dccd14d4a5641c3173819ac467ce7c 12.58MB / 15.12MB 5.3s
#9 sha256:b5d25b35c1dbfa256bea3dd164b2048d6c7f8074a555213c493c36f07bf4c559 12.58MB / 30.05MB 5.5s
#9 CANCELED

#6 [final 1/4] RUN curl -O https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm
#6 sha256:3d4cb91a064609bd3c81f84ea5ee081265336c45c2fc795064ea676ed182b978
#6 0.230   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#6 0.230                                  Dload  Upload   Total   Spent    Left  Speed
100 4071k  100 4071k    0     0   786k      0  0:00:05  0:00:05 --:--:--  858k
#6 5.479        package AWSLogsLambdaInsights-1.0-1.x86_64 is intended for a different architecture
#6 ERROR: executor failed running [/bin/sh -c curl -O $INSIGHTSEXTENSION &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm]: exit code: 1
------
 > [final 1/4] RUN curl -O https://lambda-insights-extension.s3-ap-northeast-1.amazonaws.com/amazon_linux/lambda-insights-extension.rpm &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm:
------
executor failed running [/bin/sh -c curl -O $INSIGHTSEXTENSION &&     rpm -U lambda-insights-extension.rpm &&     rm -f lambda-insights-extension.rpm]: exit code: 1
DotnetLambdaCloudwatchDashboardStack:  start: Publishing 98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53:current_account-current_region
The push refers to repository [14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1]
tag does not exist: 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53
DotnetLambdaCloudwatchDashboardStack:  fail: docker push 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53 exited with error code 1: tag does not exist: 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53
This deployment will make potentially sensitive changes according to your current security approval level (--require-approval broadening).
Please confirm you intend to make the following modifications:

IAM Statement Changes
┌───┬────────────────────────────────────────────────────┬────────┬────────────────┬──────────────────────────────┬───────────┐
│   │ Resource                                           │ Effect │ Action         │ Principal                    │ Condition │
├───┼────────────────────────────────────────────────────┼────────┼────────────────┼──────────────────────────────┼───────────┤
│ + │ ${container-image-lambda-function/ServiceRole.Arn} │ Allow  │ sts:AssumeRole │ Service:lambda.amazonaws.com │           │
└───┴────────────────────────────────────────────────────┴────────┴────────────────┴──────────────────────────────┴───────────┘
IAM Policy Changes
┌───┬────────────────────────────────────────────────┬────────────────────────────────────────────────────────────────────────────────┐
│   │ Resource                                       │ Managed Policy ARN                                                             │
├───┼────────────────────────────────────────────────┼────────────────────────────────────────────────────────────────────────────────┤
│ + │ ${container-image-lambda-function/ServiceRole} │ arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole │
└───┴────────────────────────────────────────────────┴────────────────────────────────────────────────────────────────────────────────┘
(NOTE: There may be security-related changes not in this list. See https://github.com/aws/aws-cdk/issues/1299)

Do you wish to deploy these changes (y/n)? y
DotnetLambdaCloudwatchDashboardStack: deploying... [1/1]
DotnetLambdaCloudwatchDashboardStack: creating CloudFormation changeset...
1:30:17 PM | CREATE_FAILED        | AWS::Lambda::Function      | containerimagelambdafunction210E1AAC
Resource handler returned message: "Source image 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-
west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53 does not exist. Provide a valid source image. (Service: Lambda, Statu
s Code: 400, Request ID: 8a0d1fba-5e43-4edd-85b2-aadc2af39c36)" (RequestToken: 1410134b-b227-3a06-efe7-69bdd184e12c, HandlerErrorCode: Invali
dRequest)


 ❌  DotnetLambdaCloudwatchDashboardStack failed: Error: The stack named DotnetLambdaCloudwatchDashboardStack failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Resource handler returned message: "Source image 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53 does not exist. Provide a valid source image. (Service: Lambda, Status Code: 400, Request ID: 8a0d1fba-5e43-4edd-85b2-aadc2af39c36)" (RequestToken: 1410134b-b227-3a06-efe7-69bdd184e12c, HandlerErrorCode: InvalidRequest)
    at FullCloudFormationDeployment.monitorDeployment (/opt/homebrew/lib/node_modules/aws-cdk/lib/index.js:397:10236)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deployStack2 [as deployStack] (/opt/homebrew/lib/node_modules/aws-cdk/lib/index.js:400:149585)
    at async /opt/homebrew/lib/node_modules/aws-cdk/lib/index.js:400:135508

 ❌ Deployment failed: Error: The stack named DotnetLambdaCloudwatchDashboardStack failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Resource handler returned message: "Source image 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53 does not exist. Provide a valid source image. (Service: Lambda, Status Code: 400, Request ID: 8a0d1fba-5e43-4edd-85b2-aadc2af39c36)" (RequestToken: 1410134b-b227-3a06-efe7-69bdd184e12c, HandlerErrorCode: InvalidRequest)
    at FullCloudFormationDeployment.monitorDeployment (/opt/homebrew/lib/node_modules/aws-cdk/lib/index.js:397:10236)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deployStack2 [as deployStack] (/opt/homebrew/lib/node_modules/aws-cdk/lib/index.js:400:149585)
    at async /opt/homebrew/lib/node_modules/aws-cdk/lib/index.js:400:135508

The stack named DotnetLambdaCloudwatchDashboardStack failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Resource handler returned message: "Source image 14482*****3.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-14482*****3-eu-west-1:98f861d5f5f78870913de01418525525ea5b88e4628825836a7e6550e762df53 does not exist. Provide a valid source image. (Service: Lambda, Status Code: 400, Request ID: 8a0d1fba-5e43-4edd-85b2-aadc2af39c36)" (RequestToken: 1410134b-b227-3a06-efe7-69bdd184e12c, HandlerErrorCode: InvalidRequest)

````

**Comment by srcsakthivel** [comment]
I have explicitly called out the OS Version in requirement. 

Please let me know if that looks good. Also, next if you ran into similar issue, please share the docker info as well. Thanks.

**Comment by mavi888** [comment]
Does this only works on a Architecture: x86_64? That might be the reason why it fails in my side. If that is the case, can you make this PR more generic?

**Comment by srcsakthivel** [comment]
I guess it is because of the Package - AWSLogsLambdaInsights,

$ rpm -qi AWSLogsLambdaInsights-1.0-1
Name        : AWSLogsLambdaInsights
Version     : 1.0
Release     : 1
Architecture: x86_64
Install Date: Tue 06 Jun 2023 06:25:06 PM UTC
Group       : Unspecified
Size        : 10871845
License     : Copyright Amazon, Inc.
Signature   : (none)
Source RPM  : AWSLogsLambdaInsights-1.0-1.src.rpm
Build Date  : Tue 30 May 2023 11:55:03 AM UTC
Build Host  : e02d03e374dd
Relocations : (not relocatable)
URL         : https://code.amazon.com/packages/AWSLogsLambdaInsights
Summary     : Package to test the RPM transform in the BuildArtifactTransformService.
Description :
Package to test the RPM transform in the BuildArtifactTransformService.

Other details: https://github.com/awsdocs/amazon-cloudwatch-user-guide/issues/97

**Comment by srcsakthivel** [comment]
@mavi888 - just following up to see if you have any updates.

**Comment by jbesw** [comment]
Due to PTO, we will pick this up for review in about 2 weeks. Sorry for the delay.

**Comment by mavi888** [comment]
The layer is also available for ARN: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-Getting-Started-docker.html#Lambda-Insights-Getting-Started-docker-ARM64

Can you make it more generic? So everybody can deploy it?

**Comment by srcsakthivel** [comment]
@mavi888 - I have updated the code with variables. Updated the Readme.md as well for the ARM.

**Comment by mavi888** [comment]
Do you also need to add instructions to the readme on how to work with these variables?

**Comment by srcsakthivel** [comment]
I have added those.

**Comment by mavi888** [comment]
Hello, I tested this and I cannot get the dashboard to show metrics.
There are no errors in deployments and I'm using the ARM64 configuration.


Screenshot 2023-08-07 at 10 17 38

**Comment by jbesw** [comment]
This submission is too complex for patterns. Once the code is working, I would recommend resubmitting to the Serverless Land Repos collection: https://serverlessland.com/repos.

**Pull request closed without merging**